include/lo2s/monitor/dummy_monitor.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of the lo2s software.
+ * Linux OTF2 sampling
+ *
+ * Copyright (c) 2016,
+ *    Technische Universitaet Dresden, Germany
+ *
+ * lo2s is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * lo2s is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with lo2s.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <lo2s/process_map/process.hpp>
+#include <lo2s/process_map/thread.hpp>
+
+#include  <string>
+
+extern "C" {
+#include <sys/types.h>
+}
+
+namespace lo2s
+{
+
+namespace monitor
+{
+
+class DummyMonitor
```

Comment:
Please split this class in a pure virtual `AbstractProcessMonitor` and a dummy implementation `DummyProcessMonitor`. That way we won't get silent issues if we change/add methods to the base class of the actual `ProcessMonitor`.